### PR TITLE
Fix n+1 on /adoptable_pets

### PR DESCRIPTION
**Pull request by albertchae**
### Describe your change in plain English.

These 2 queries have N+1s
`AdopterApplication Load (0.1ms)  SELECT "adopter_applications".* FROM "adopter_applications" WHERE "adopter_applications"."pet_id" = $1  [["pet_id", 8]]`

`ActiveStorage::Attachment Exists? (0.2ms)  SELECT 1 AS one FROM "active_storage_attachments" WHERE "active_storage_attachments"."record_id" = $1 AND "active_storage_attachments"."record_type" = $2 AND "active_storage_attachments"."name" = $3 LIMIT $4  [["record_id", 9], ["record_type", "Pet"], ["name", "images"], ["LIMIT", 1]]`

The latter is using a `has_many_attached`, so we have to name the eager load slightly differently https://blog.saeloun.com/2020/03/06/eagerload-active-storage-models/

### Link to the issue

No issue required per @edwinthinks

### CODE REVIEW

**Review by Thrillberg — APPROVED** [review]
Thanks!! Looks great!

**Review by edwinthinks — APPROVED** [review]
[https://lgtm.us](https://lgtm.us)
